Makefile

```diff
@@ -40,7 +40,8 @@ GOTOOL            ?= go tool
 GOTEST            ?= gotestsum --junitfile $(TEST_OUTPUT_DIR)/$(KREPO)-unit-tests.xml --format pkgname-and-test-fails --
 
 GOPKGS             = ./cmd/... ./pkg/apis/... ./pkg/function/... ./pkg/routing/... ./pkg/sources/... ./pkg/targets/... ./pkg/transformation/...
-LDFLAGS            = -extldflags=-static -w -s
+LDFLAGS            = -w -s
```

Comment:
We do want binaries built by `make build` to be static, don't we? (even though we don't really ever build binaries that way)

Reply:
100% yes https://github.com/triggermesh/triggermesh/pull/311/commits/1fa20ae18eb05cd7c18162059a707137b6770263

Reply:
The referenced commit seems to **remove** "-static" for all binaries (except 2 of them)

Reply:
maybe there is a way of writing this more clear, this is the intention:

`LDFLAGS` contains flags that are common to all builds
`LDFLAGS_STATIC` is to be used for those that should be linked statically (all but xslttransform)

https://github.com/triggermesh/triggermesh/blob/1fa20ae18eb05cd7c18162059a707137b6770263/Makefile#L43-L44

All builds but `confluenttarget-adapter` and `xslttransform-adapter` use the static flavor without the need for C libraries.

https://github.com/triggermesh/triggermesh/blob/1fa20ae18eb05cd7c18162059a707137b6770263/Makefile#L71-L72

The confluent adapter uses the static flavor also, but includes C library references

https://github.com/triggermesh/triggermesh/blob/1fa20ae18eb05cd7c18162059a707137b6770263/Makefile#L74-L75

And finally, this very special `xslttransform` uses functions at the glibc library that seems to depend on the OS and cannot be statically linked. For this one is the only adapter using the non static linking and the C library references.

https://github.com/triggermesh/triggermesh/blob/1fa20ae18eb05cd7c18162059a707137b6770263/Makefile#L77-L79

This is the (truncated :carpentry_saw:) output I get when building

```
go build -ldflags "-w -s -extldflags=-static" -o /home/pablo/go/src/github.com/triggermesh/triggermesh/_output/salesforcetarget-adapter ./cmd/salesforcetarget-adapter                        
CGO_ENABLED=1 go build -ldflags "-w -s" -o /home/pablo/go/src/github.com/triggermesh/triggermesh/_output/xslttransform-adapter ./cmd/xslttransform-adapter                                    
go build -ldflags "-w -s -extldflags=-static" -o /home/pablo/go/src/github.com/triggermesh/triggermesh/_output/datadogtarget-adapter ./cmd/datadogtarget-adapter                              
go build -ldflags "-w -s -extldflags=-static" -o /home/pablo/go/src/github.com/triggermesh/triggermesh/_output/googlecloudworkflowstarget-adapter ./cmd/googlecloudworkflowstarget-adapter    
go build -ldflags "-w -s -extldflags=-static" -o /home/pablo/go/src/github.com/triggermesh/triggermesh/_output/awslambdatarget-adapter ./cmd/awslambdatarget-adapter                          
go build -ldflags "-w -s -extldflags=-static" -o /home/pablo/go/src/github.com/triggermesh/triggermesh/_output/splitter-adapter ./cmd/splitter-adapter                                        
go build -ldflags "-w -s -extldflags=-static" -o /home/pablo/go/src/github.com/triggermesh/triggermesh/_output/filter-adapter ./cmd/filter-adapter                                            
go build -ldflags "-w -s -extldflags=-static" -o /home/pablo/go/src/github.com/triggermesh/triggermesh/_output/slacktarget-adapter ./cmd/slacktarget-adapter                                  
go build -ldflags "-w -s -extldflags=-static" -o /home/pablo/go/src/github.com/triggermesh/triggermesh/_output/triggermesh-webhook ./cmd/triggermesh-webhook                                  
go build -ldflags "-w -s -extldflags=-static" -o /home/pablo/go/src/github.com/triggermesh/triggermesh/_output/hasuratarget-adapter ./cmd/hasuratarget-adapter                                
go build -ldflags "-w -s -extldflags=-static" -o /home/pablo/go/src/github.com/triggermesh/triggermesh/_output/httptarget-adapter ./cmd/httptarget-adapter                                    
go build -ldflags "-w -s -extldflags=-static" -o /home/pablo/go/src/github.com/triggermesh/triggermesh/_output/oracletarget-adapter ./cmd/oracletarget-adapter                                
go build -ldflags "-w -s -extldflags=-static" -o /home/pablo/go/src/github.com/triggermesh/triggermesh/_output/awscognitoidentitysource-adapter ./cmd/awscognitoidentitysource-adapter        
go build -ldflags "-w -s -extldflags=-static" -o /home/pablo/go/src/github.com/triggermesh/triggermesh/_output/elasticsearchtarget-adapter ./cmd/elasticsearchtarget-adapter                  
go build -ldflags "-w -s -extldflags=-static" -o /home/pablo/go/src/github.com/triggermesh/triggermesh/_output/azureservicebussource-adapter ./cmd/azureservicebussource-adapter              
go build -ldflags "-w -s -extldflags=-static" -o /home/pablo/go/src/github.com/triggermesh/triggermesh/_output/sendgridtarget-adapter ./cmd/sendgridtarget-adapter                            
go build -ldflags "-w -s -extldflags=-static" -o /home/pablo/go/src/github.com/triggermesh/triggermesh/_output/salesforcesource-adapter ./cmd/salesforcesource-adapter                        
go build -ldflags "-w -s -extldflags=-static" -o /home/pablo/go/src/github.com/triggermesh/triggermesh/_output/azurequeuestoragesource-adapter ./cmd/azurequeuestoragesource-adapter          
go build -ldflags "-w -s -extldflags=-static" -o /home/pablo/go/src/github.com/triggermesh/triggermesh/_output/splunktarget-adapter ./cmd/splunktarget-adapter                                
go build -ldflags "-w -s -extldflags=-static" -o /home/pablo/go/src/github.com/triggermesh/triggermesh/_output/logztarget-adapter ./cmd/logztarget-adapter                                    
go build -ldflags "-w -s -extldflags=-static" -o /home/pablo/go/src/github.com/triggermesh/triggermesh/_output/opentelemetrytarget-adapter ./cmd/opentelemetrytarget-adapter                  
go build -ldflags "-w -s -extldflags=-static" -o /home/pablo/go/src/github.com/triggermesh/triggermesh/_output/slacksource-adapter ./cmd/slacksource-adapter                                  
go build -ldflags "-w -s -extldflags=-static" -o /home/pablo/go/src/github.com/triggermesh/triggermesh/_output/jiratarget-adapter ./cmd/jiratarget-adapter                                    
CGO_ENABLED=1 go build -ldflags "-w -s -extldflags=-static" -o /home/pablo/go/src/github.com/triggermesh/triggermesh/_output/confluenttarget-adapter ./cmd/confluenttarget-adapter 
go build -ldflags "-w -s -extldflags=-static" -o /home/pablo/go/src/github.com/triggermesh/triggermesh/_output/tektontarget-adapter ./cmd/tektontarget-adapter
go build -ldflags "-w -s -extldflags=-static" -o /home/pablo/go/src/github.com/triggermesh/triggermesh/_output/ocimetricssource-adapter ./cmd/ocimetricssource-adapter
go build -ldflags "-w -s -extldflags=-static" -o /home/pablo/go/src/github.com/triggermesh/triggermesh/_output/azureeventhubsource-adapter ./cmd/azureeventhubsource-adapter
go build -ldflags "-w -s -extldflags=-static" -o /home/pablo/go/src/github.com/triggermesh/triggermesh/_output/infratarget-adapter ./cmd/infratarget-adapter
go build -ldflags "-w -s -extldflags=-static" -o /home/pablo/go/src/github.com/triggermesh/triggermesh/_output/alibabaosstarget-adapter ./cmd/alibabaosstarget-adapter
go build -ldflags "-w -s -extldflags=-static" -o /home/pablo/go/src/github.com/triggermesh/triggermesh/_output/awskinesissource-adapter ./cmd/awskinesissource-adapter
go build -ldflags "-w -s -extldflags=-static" -o /home/pablo/go/src/github.com/triggermesh/triggermesh/_output/transformation-adapter ./cmd/transformation-adapter
go build -ldflags "-w -s -extldflags=-static" -o /home/pablo/go/src/github.com/triggermesh/triggermesh/_output/twiliosource-adapter ./cmd/twiliosource-adapter
go build -ldflags "-w -s -extldflags=-static" -o /home/pablo/go/src/github.com/triggermesh/triggermesh/_output/zendesktarget-adapter ./cmd/zendesktarget-adapter
go build -ldflags "-w -s -extldflags=-static" -o /home/pablo/go/src/github.com/triggermesh/triggermesh/_output/googlecloudstoragetarget-adapter ./cmd/googlecloudstoragetarget-adapter
go build -ldflags "-w -s -extldflags=-static" -o /home/pablo/go/src/github.com/triggermesh/triggermesh/_output/azureiothubsource-adapter ./cmd/azureiothubsource-adapter
go build -ldflags "-w -s -extldflags=-static" -o /home/pablo/go/src/github.com/triggermesh/triggermesh/_output/awscloudwatchlogssource-adapter ./cmd/awscloudwatchlogssource-adapter
go build -ldflags "-w -s -extldflags=-static" -o /home/pablo/go/src/github.com/triggermesh/triggermesh/_output/triggermesh-controller ./cmd/triggermesh-controller
go build -ldflags "-w -s -extldflags=-static" -o /home/pablo/go/src/github.com/triggermesh/triggermesh/_output/awssnstarget-adapter ./cmd/awssnstarget-adapter
go build -ldflags "-w -s -extldflags=-static" -o /home/pablo/go/src/github.com/triggermesh/triggermesh/_output/awscodecommitsource-adapter ./cmd/awscodecommitsource-adapter
go build -ldflags "-w -s -extldflags=-static" -o /home/pablo/go/src/github.com/triggermesh/triggermesh/_output/awssqstarget-adapter ./cmd/awssqstarget-adapter
go build -ldflags "-w -s -extldflags=-static" -o /home/pablo/go/src/github.com/triggermesh/triggermesh/_output/googlecloudpubsubsource-adapter ./cmd/googlecloudpubsubsource-adapter
go build -ldflags "-w -s -extldflags=-static" -o /home/pablo/go/src/github.com/triggermesh/triggermesh/_output/twiliotarget-adapter ./cmd/twiliotarget-adapter
go build -ldflags "-w -s -extldflags=-static" -o /home/pablo/go/src/github.com/triggermesh/triggermesh/_output/azureeventhubstarget-adapter ./cmd/azureeventhubstarget-adapter
go build -ldflags "-w -s -extldflags=-static" -o /home/pablo/go/src/github.com/triggermesh/triggermesh/_output/awscloudwatchsource-adapter ./cmd/awscloudwatchsource-adapter
go build -ldflags "-w -s -extldflags=-static" -o /home/pablo/go/src/github.com/triggermesh/triggermesh/_output/awss3target-adapter ./cmd/awss3target-adapter
go build -ldflags "-w -s -extldflags=-static" -o /home/pablo/go/src/github.com/triggermesh/triggermesh/_output/awsdynamodbtarget-adapter ./cmd/awsdynamodbtarget-adapter
go build -ldflags "-w -s -extldflags=-static" -o /home/pablo/go/src/github.com/triggermesh/triggermesh/_output/googlecloudfirestoretarget-adapter ./cmd/googlecloudfirestoretarget-adapter
go build -ldflags "-w -s -extldflags=-static" -o /home/pablo/go/src/github.com/triggermesh/triggermesh/_output/awssnssource-adapter ./cmd/awssnssource-adapter
go build -ldflags "-w -s -extldflags=-static" -o /home/pablo/go/src/github.com/triggermesh/triggermesh/_output/awssqssource-adapter ./cmd/awssqssource-adapter
go build -ldflags "-w -s -extldflags=-static" -o /home/pablo/go/src/github.com/triggermesh/triggermesh/_output/awscomprehendtarget-adapter ./cmd/awscomprehendtarget-adapter
go build -ldflags "-w -s -extldflags=-static" -o /home/pablo/go/src/github.com/triggermesh/triggermesh/_output/awsdynamodbsource-adapter ./cmd/awsdynamodbsource-adapter
go build -ldflags "-w -s -extldflags=-static" -o /home/pablo/go/src/github.com/triggermesh/triggermesh/_output/awskinesistarget-adapter ./cmd/awskinesistarget-adapter
go build -ldflags "-w -s -extldflags=-static" -o /home/pablo/go/src/github.com/triggermesh/triggermesh/_output/uipathtarget-adapter ./cmd/uipathtarget-adapter
go build -ldflags "-w -s -extldflags=-static" -o /home/pablo/go/src/github.com/triggermesh/triggermesh/_output/googlesheettarget-adapter ./cmd/googlesheettarget-adapter
go build -ldflags "-w -s -extldflags=-static" -o /home/pablo/go/src/github.com/triggermesh/triggermesh/_output/awsperformanceinsightssource-adapter ./cmd/awsperformanceinsightssource-adapter
go build -ldflags "-w -s -extldflags=-static" -o /home/pablo/go/src/github.com/triggermesh/triggermesh/_output/webhooksource-adapter ./cmd/webhooksource-adapter
go build -ldflags "-w -s -extldflags=-static" -o /home/pablo/go/src/github.com/triggermesh/triggermesh/_output/zendesksource-adapter ./cmd/zendesksource-adapter
go build -ldflags "-w -s -extldflags=-static" -o /home/pablo/go/src/github.com/triggermesh/triggermesh/_output/httppollersource-adapter ./cmd/httppollersource-adapter
go build -ldflags "-w -s -extldflags=-static" -o /home/pablo/go/src/github.com/triggermesh/triggermesh/_output/awscognitouserpoolsource-adapter ./cmd/awscognitouserpoolsource-adapter
```

So maybe I am missing something?
or should I change the makefile to use a different pattern that does not lead to confussion, for example instead of having variables `LDFLAGS` and `LDFLAGS_STATIC`, renaming to `LDFLAGS` (which contains static) and `LDFLAGS_NON_STATIC` for the only exception that we have?

Reply:
Ah, sorry, that's my bad, I didn't quite understand what this "filter-out" function was doing!